Remove `WKBundlePageGetMainFrame` use in `InjectedBundlePage::shouldInsertNode`

charliewolfe opened the pull request:
#### 09a39c7410120fcf29cdb6310eabad53edaba847
<pre>
Remove `WKBundlePageGetMainFrame` use in `InjectedBundlePage::shouldInsertNode`
<a href="https://bugs.webkit.org/show_bug.cgi?id=282856">https://bugs.webkit.org/show_bug.cgi?id=282856</a>
<a href="https://rdar.apple.com/139541150">rdar://139541150</a>

Reviewed by Wenson Hsieh.

Using `WKBundlePageGetMainFrame` when the main frame is out-of-process leads to a crash. We should use
the frame associated with the `WKBundleNodeHandleRef` instead.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleNodeHandle.cpp:
(WKBundleNodeHandleCopyOwningDocumentFrame):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleNodeHandlePrivate.h:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::dumpPath):
(WTR::string):
(WTR::InjectedBundlePage::shouldInsertNode):

Canonical link: <a href="https://commits.webkit.org/286372@main">https://commits.webkit.org/286372@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3c6025262e349617d7e45b6c427e9d0db7f0e620

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75808 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/54837 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/28701 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80299 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27074 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/77924 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/63979 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3131 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/59441 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17607 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78875 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/63979 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65104 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39800 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/63979 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22587 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/25401 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/63979 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22925 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/81769 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3177 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/1990 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67669 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3331 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65071 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66974 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10920 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/9047 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11709 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3127 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/5948 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3148 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/4086 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/3155 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleNodeHandle.cpp:
```diff
@@ -232,3 +232,10 @@ void WKBundleNodeHandleSetHTMLInputElementAutoFillButtonEnabled(WKBundleNodeHand
     // FIXME: Would put ASSERT_NOT_REACHED() here but some compilers are warning the function is "noreturn".
 }
+
+WKBundleFrameRef WKBundleNodeHandleDocumentFrame(WKBundleNodeHandleRef documentHandleRef)
```
Since this returns a +1 object (using `leakRef()`), the caller should use `adoptWK` to wrap it in a smart pointer, to prevent a leak.

We usually denote this behavior by using the name `Copy` in the function itself (like `WKBundleNodeHandleCopyDocumentFrame`), but in this case, that symbol already exists (and behaves slightly differently, requiring the node handle to be a document itself).

WDYT about renaming to something like `WKBundleNodeHandleCopyOwningDocumentFrame`, and also adding `adoptWK` at the call site?

charliewolfe committed:
Remove `WKBundlePageGetMainFrame` use in `InjectedBundlePage::shouldInsertNode`
https://bugs.webkit.org/show_bug.cgi?id=282856
rdar://139541150

Reviewed by Wenson Hsieh.

Using `WKBundlePageGetMainFrame` when the main frame is out-of-process leads to a crash. We should use
the frame associated with the `WKBundleNodeHandleRef` instead.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleNodeHandle.cpp:
(WKBundleNodeHandleCopyOwningDocumentFrame):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleNodeHandlePrivate.h:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::dumpPath):
(WTR::string):
(WTR::InjectedBundlePage::shouldInsertNode):

Canonical link: https://commits.webkit.org/286372@main
